bentoml/yatai/client/bento_repository_api.py

```diff
@@ -344,62 +346,79 @@ def list(
+        >>> yatai_client.repository.delete(labels='ci=failed, cohort=20')
         """
-        track('py-api-prune')
-        list_bentos_result = self.list(bento_name=bento_name, labels=labels,)
-        if list_bentos_result.status.status_code != yatai_proto.status_pb2.Status.OK:
-            error_code, error_message = status_pb_to_error_code_and_message(
-                list_bentos_result.status
+        track('py-api-delete')
+        bento_delete_list = []
+        if prune is True:
+            bento_delete_list = self.list()
+            logger.info(f'Deleting all {len(bento_delete_list)} BentoML saved bundles.')
+        else:
+            if bento_tag:
+                bento_delete_list.append(self.get(bento_tag))
+                logger.info(f'Deleting saved Bento bundle {bento_tag}')
+            if bento_name is not None and bento_version is not None:
+                # ignore labels when version is provided.
+                bento_delete_list.append(self.get(f'{bento_name}:{bento_version}'))
+                logger.info(f'Deleting saved Bento bundle {bento_name}:{bento_version}')
+            else:
+                bento_delete_list = self.list(bento_name=bento_name, labels=labels)
+                log_message = (
+                    f'Deleting all {len(bento_delete_list)} saved Bento bundles'
+                )
+                if bento_name is not None:
+                    log_message += f' with name: {bento_name},'
+                if labels is not None:
+                    log_message += f' with labels match to {labels}'
+                logger.info(log_message)
+        for bento in bento_delete_list:
+            if require_confirm and not click.confirm(
+                f'Permanently delete {bento.name}:{bento.version}?'
+            ):
+                return
+            result = self.yatai_service.DangerouslyDeleteBento(
+                DangerouslyDeleteBentoRequest(
+                    bento_name=bento.name, bento_version=bento.version
+                )
             )
-            raise BentoMLException(f'{error_code}:{error_message}')
-        for bento in list_bentos_result.bentos:
-            bento_tag = f'{bento.name}:{bento.version}'
-            try:
-                self.delete(bento_tag)
-            except BentoMLException as e:
-                logger.error(f'Failed to delete Bento {bento_tag}: {e}')
+            if result.status.status_code != yatai_proto.status_pb2.Status.OK:
+                error_code, error_message = status_pb_to_error_code_and_message(
+                    result.status
+                )
+                # Rather than raise Exception, continue to delete the next bentos
+                logger.error(
+                    f'Failed to delete {bento.name}:{bento.version} '
+                    f'{error_code}:{error_message}'
+                )
+            logger.info(f'Deleted {bento.name}:{bento.version}')
```

Comment:
this should be in an `else` clause right? otherwise, it will print this even if the `if result.status.status_code != yatai_proto.status_pb2.Status.OK`

Reply:
add a test for this.